Audit changes and use latest theme

emclaug2 opened the pull request:
<!-- Include a bulleted list summarizing the main changes you have made in this PR -->
#### Changes proposed in this Pull Request:

Use latest theme package @^5.2.0

Channel Value Changes:
-  Change `full config` icon color to red[500].

Empty State Changes:
- Made action button outlined variant in both button examples

InfoListItem Changes:
- Font and Bg color changes to be more app-reaslistic.

Scorecard Changes:
- Replace all red[500] bg color headers with blue[500]

joebochill commented:
Same comments as a dropped on the React PR (hey, the demos are the same! 😄 ), plus:

- Drawer Rail example should default to _not_ condensed
- ILI full list...leaf is not an icon of temperature

emclaug2 commented:
I'm working on matching React's comments right now 🗡️ 

>>Drawer Rail example should default to not condensed

This was changed to prevent this from happening: 
![nav-item-tooltip-issue](https://user-images.githubusercontent.com/6538289/105036991-4ef98100-5a2b-11eb-8245-a7d8f7355104.gif)

joebochill commented:
> I'm working on matching React's comments right now 🗡️
> 
> > > Drawer Rail example should default to not condensed
> 
> This was changed to prevent this from happening:
> 
> ![nav-item-tooltip-issue](https://user-images.githubusercontent.com/6538289/105036991-4ef98100-5a2b-11eb-8245-a7d8f7355104.gif)
> 
> I blame storybook because I was never able to reproduce this outside of storybook.

Lol what the heck

joebochill commented:
> > > Drawer Rail example should default to not condensed
> 
> This was changed to prevent this from happening:
> 
> I blame storybook because I was never able to reproduce this outside of storybook.

Lol what the heck

emclaug2 commented:
This PR is ready for another round of review; there are some outstanding RTL issues in the UserMenu and DropdownToolbar that I'm working on right now.

I think the RTL stuff in our overlays are a bit funky right now.

https://pxblue-angular-library.web.app/

emclaug2 commented:
Here's a summary of how our components currently handle icon-inversion in RTL.    At some point all components were set up to do the inversion but this logic was removed.  If this is looking fine then I'm going to manually add those RTL inversions to all required storybook stories. 

`<pxb-hero>`: no-flip
`<pxb-channel-value>`: no-flip
`<pxb-drawer>`: flip
`<pxb-drawer> (rail)` no-flip
`<pxb-info-list-item>`: flip
`<pxb-empty-state>`: no-flip
`<pxb-score-card>`: no-flip

huayunh commented:
Yeah that was my fault — only icons indicating directionalities should be flipped. 

See https://px-blue.github.io/patterns/internationalization#bidirectionality--right-to-left-(rtl)-support for what to flip

huayunh commented:
So essentially when a user passes in a custom icon, it is the user's job to flip icons, not that of the component library. 

Icons that are part of the component library should be flipped on our behalf. For example, the chevron on an Info List Item

emclaug2 commented:
> So essentially when a user passes in a custom icon, it is the user's job to flip icons, not that of the component library.
> 
> Icons that are part of the component library should be flipped on our behalf. For example, the chevron on an Info List Item

I'm okay with this idea as long as are consistent; I think we should take out the icon flipping logic from the InfoListItem component (that'll remove it from Drawer too).

joebochill commented:
> So essentially when a user passes in a custom icon, it is the user's job to flip icons, not that of the component library.
> 
> Icons that are part of the component library should be flipped on our behalf. For example, the chevron on an Info List Item

This is correct and it's how we are doing it in react right now. We can't auto-flip icons that are passed in by the user because we don't know what they are (they may or may not need to be flipped). The one's we use internally, we can flip because we know what they are. I believe there might be an older issue already logged about this.

emclaug2 commented:
This PR also fixes a RTL menu-icon spacing bug in the DropdownToolbar.  

Stories should match between React and Angular pretty closely now.  Biggest differences are `full config` and `nested nav items` stories for the Drawer.  

React's nested nav item story shows an example of 4 levels of nav items which isn't something we should be encouraging users to do, so I left Angular the way it was.

emclaug2 commented:
Hosted here: https://pxblue-angular-library.web.app/

emclaug2 commented:
This PR also has the InfoListItem RTL inversion logic removed (which is in a separate PR).

huayunh commented:
https://pxblue-angular-library.web.app/?path=/story/components-channel-value--with-full-config
Flip icon

huayunh commented:
https://pxblue-angular-library.web.app/?path=/story/components-drawer--with-basic-usage
Identity management — use the solid icon, not the outlined version

huayunh commented:
Same for https://pxblue-angular-library.web.app/?path=/story/components-drawer--with-nested-nav-items

huayunh commented:
I suppose this is a theme problem but the selected nav item should turn blue (for both icons and labels)
![image](https://user-images.githubusercontent.com/8997218/105419852-5c6e6100-5c0d-11eb-8195-ce5fc9249af7.png)

huayunh commented:
https://pxblue-angular-library.web.app/?path=/story/components-empty-state--with-full-config 
The size of the button differs between angular and React... pick your favorite one :P

huayunh commented:
Make a story or log a bug to make Mobile Stepper left-to-right friendly

huayunh commented:
https://pxblue-angular-library.web.app/?path=/story/components-user-menu--with-basic-usage
![image](https://user-images.githubusercontent.com/8997218/105420501-79576400-5c0e-11eb-95d4-1603a06cd2ca.png)
* React is using font weight 400 here
* Flip the logout icon

huayunh commented:
https://pxblue-angular-library.web.app/?path=/story/components-user-menu--within-a-toolbar
version number ought to be pinned to the other side
![image](https://user-images.githubusercontent.com/8997218/105420649-bae80f00-5c0e-11eb-8233-30fb6c936f1e.png)

huayunh commented:
Is ng dark theme still due for an update at some point? Right now the dropdown toolbar shares the same color as its background under dark theme
![image](https://user-images.githubusercontent.com/8997218/105484531-7b0a4180-5c79-11eb-9301-6013b75deb7f.png)

huayunh commented:
button text and icon are not perfectly aligned

![image](https://user-images.githubusercontent.com/8997218/105484770-e0f6c900-5c79-11eb-925e-02b360c8b93e.png)

Need a 
```css
{
    display: flex;
    align-items: center;
}
```
somewhere for the `.mat-button-wrapper`

And change icon font size to be 20px instead of 24px

joebochill commented:
> Is ng dark theme still due for an update at some point? Right now the dropdown toolbar shares the same color as its background under dark theme

Yeah, we have an upcoming story to address dark theme changes once we do an audit.

emclaug2 commented:
> https://pxblue-angular-library.web.app/?path=/story/components-empty-state--with-full-config
> The size of the button differs between angular and React... pick your favorite one :P

I've remove React's `small` prop from its button.

emclaug2 commented:
This is updated now: https://pxblue-angular-library.web.app/?path=/story/*

Empty State add icon is 20px now.
Icon and Text in EmptyState action buttons content is vertically centered. (component change)
UserMenu icons are RTL-inverted.
UserMenu list item font weight is 400 now. (component change)

huayunh commented:
![image](https://user-images.githubusercontent.com/8997218/105526057-e66f0600-5caf-11eb-87f3-d95051648d58.png)

Do we want to pick a circular icon (like `voltage_circled_outline`) so that the overflowing icon issue is less obvious?

huayunh commented:
Use a better subheader component here
![image](https://user-images.githubusercontent.com/8997218/105526544-92b0ec80-5cb0-11eb-96f7-ff01f0710617.png)

And change the icon of the identity management to be solid

huayunh commented:
If you want to go fancy on drawer subheader... instead of doing what React is doing, you can try something like this
![image](https://user-images.githubusercontent.com/8997218/105526975-1bc82380-5cb1-11eb-9507-d6aaba60fcb6.png)
or do an organization switcher
![image](https://user-images.githubusercontent.com/8997218/105527238-706b9e80-5cb1-11eb-93d9-59e44be3717d.png)
that would trigger a dropdown to select an org / a location

emclaug2 commented:
> If you want to go fancy on drawer subheader... instead of doing what React is doing, you can try something like this
> ![image](https://user-images.githubusercontent.com/8997218/105526975-1bc82380-5cb1-11eb-9507-d6aaba60fcb6.png)
> or do an organization switcher
> ![image](https://user-images.githubusercontent.com/8997218/105527238-706b9e80-5cb1-11eb-93d9-59e44be3717d.png)
> that would trigger a dropdown to select an org / a location

I think this is better as a separate issue since it'd have to be done for angular and react.

emclaug2 commented:
Ok latest changes are live here:

- Drawer person icon should be filled now.
- Barley icon replaced with better, rounder icon
- Drawer subheader looks better (matching react)

huayunh commented:
https://pxblue-angular-library.web.app/?path=/story/components-drawer--with-full-config
Update the menu items so that they make more sense (match previous stories)

huayunh commented:
https://pxblue-angular-library.web.app/?path=/story/components-info-list-item--with-full-config
![image](https://user-images.githubusercontent.com/8997218/105588882-a36c6b80-5d5f-11eb-8e9e-1f6203d5a70e.png)
Left icon not center-aligned

Same for the first info list item of https://pxblue-angular-library.web.app/?path=/story/components-info-list-item--within-a-full-list
![image](https://user-images.githubusercontent.com/8997218/105589685-d9a9eb00-5d5f-11eb-827d-b33ed20e5623.png)
All the other icons of ILI appear fine to me

huayunh commented:
https://pxblue-angular-library.web.app/?path=/story/components-score-card--with-full-config

![image](https://user-images.githubusercontent.com/8997218/105590711-22fa3a80-5d60-11eb-8d5b-d37550f75c6f.png)
Flip the list icon, or replace it with `favorite` or `bookmark` (which are two more commonly used icons for scorecard)

huayunh commented:
Don't know if this is a storybook issue or ACL issue —
![image](https://user-images.githubusercontent.com/8997218/105593358-debb6a00-5d60-11eb-9b29-0f1c9600486c.png)

Step to reproduce: 
1. Go to https://pxblue-angular-library.web.app/?path=/story/components-user-menu--within-a-toolbar  (Or any user menu story)
2. Without touching anything else, toggle the i18n on so that it reads RTL
3. Now click the avatar. Observe what you see in the screenshot

Note that this would not occur if you enter a story with i18n already turned on (i.e., turn on i18n and switch between different stories)
This would not happen either if you click the avatar first, and then toggle i18n

huayunh commented:
Shrink the user menu footer height in the within a toolbar story.
React does 40px tall
![image](https://user-images.githubusercontent.com/8997218/105595410-6bfebe80-5d61-11eb-8f14-ee065b8d205b.png)

huayunh commented:
I noticed that there is a 4px left margin on `.pxb-user-menu-header-avatar`. This caused misalignment of several UI elements: 

![image](https://user-images.githubusercontent.com/8997218/105598746-50e07e80-5d62-11eb-9241-11187b1529cb.png)

huayunh commented:
Oh and can we give the user menu an 8dp shadow?

emclaug2 commented:
> https://pxblue-angular-library.web.app/?path=/story/components-score-card--with-full-config
> 
> ![image](https://user-images.githubusercontent.com/8997218/105590711-22fa3a80-5d60-11eb-8d5b-d37550f75c6f.png)
> Flip the list icon, or replace it with `favorite` or `bookmark` (which are two more commonly used icons for scorecard)
> 
> I might let the React side slipped... please double check

I'm just going to replace the list icon.   RTL-sensitive icons have no place in these demos since they're too difficult to manage.

emclaug2 commented:
This PR now fixes #202 

- Fixed Drawer shadow, now uses mat-elevation-z4
- Added shadow to UserMenu menu (mat-elevation-z8)
- Removed 4px offset margin to UserMenu header avatar

Available here: https://pxblue-angular-library.web.app

emclaug2 commented:
> https://pxblue-angular-library.web.app/?path=/story/components-info-list-item--with-full-config
> ![image](https://user-images.githubusercontent.com/8997218/105588882-a36c6b80-5d5f-11eb-8e9e-1f6203d5a70e.png)
> Left icon not center-aligned
> 
> Same for the first info list item of https://pxblue-angular-library.web.app/?path=/story/components-info-list-item--within-a-full-list
> ![image](https://user-images.githubusercontent.com/8997218/105589685-d9a9eb00-5d5f-11eb-827d-b33ed20e5623.png)
> All the other icons of ILI appear fine to me

I don't think this is a component issue, I could add an offset margin to the image to straighten it up.

joebochill commented:
> I don't think this is a component issue, I could add an offset margin to these problem icons to straighten them.

It definitely looks like a component issue, but related to using `<i>` for icons instead of `<mat-icon>`. I'd be happy if we just swap those out.

joebochill commented:
Changes look excellent to me 👍 

The only thing I noticed seemed to be a slight inconsistency with the divider props on the drawers being on by default sometimes and off others.

emclaug2 commented:
> Changes look excellent to me 👍
> 
> The only thing I noticed seemed to be a slight inconsistency with the divider props on the drawers being on by default sometimes and off others.

I think nested nav item story is the only instance that has the dividers on by default;  pretty sure this was intentional to match react + better show nav depth.

emclaug2 commented:
> > I don't think this is a component issue, I could add an offset margin to these problem icons to straighten them.
> 
> It definitely looks like a component issue, but related to using `<i>` for icons instead of `<mat-icon>`. I'd be happy if we just swap those out.

Looking closer the icon container is centered, but the icon itself looks shifted up.   I can swap this out with a different mat-icon but it won't match react anymore. 

![image](https://user-images.githubusercontent.com/6538289/105856691-80d48f80-5fb7-11eb-817b-eae7506db46b.png)


I'll swap this icon out with a mat-icon.

emclaug2 commented:
`with full config` and `within a list` uses devices icon instead of the off-centered icon from before. 

https://pxblue-angular-library.web.app/?path=/story/components-info-list-item--with-full-config

If this is looking good let's get this thing merged in.

huayunh commented:
This is not fixed or logged
https://github.com/pxblue/angular-component-library/pull/210#issuecomment-766091719

huayunh commented:
https://pxblue-angular-library.web.app/?path=/story/components-info-list-item--within-a-full-list
Use an actual temperature icon
![image](https://user-images.githubusercontent.com/8997218/105863450-e8daa400-5fbe-11eb-855a-1520f66d3e7b.png)

emclaug2 commented:
> This is not fixed or logged
> [#210 (comment)](https://github.com/pxblue/angular-component-library/pull/210#issuecomment-766091719)

This is a RTL issue in the storybook addon; RTL dir attribute not being set correctly in this case. 

https://github.com/pxblue/storybook-addons/issues/15

emclaug2 commented:
> https://pxblue-angular-library.web.app/?path=/story/components-info-list-item--within-a-full-list
> Use an actual temperature icon
> ![image](https://user-images.githubusercontent.com/8997218/105863450-e8daa400-5fbe-11eb-855a-1520f66d3e7b.png)

Temperature icon added: https://pxblue-angular-library.web.app/?path=/story/components-info-list-item--within-a-full-list

huayunh commented:
dark theme rail footer, try using the eaton logo from our logo library

https://pxblue-angular-library.web.app/?path=/story/components-drawer--with-a-nav-rail

![image](https://user-images.githubusercontent.com/8997218/105867068-a61acb00-5fc2-11eb-9acf-ca237b6510bf.png)

huayunh commented:
https://pxblue-angular-library.web.app/?path=/story/components-score-card--with-score-badge

dark theme scorecard, make the badge background to match the card background
![image](https://user-images.githubusercontent.com/8997218/105867480-26d9c700-5fc3-11eb-95e8-a5ff20536ed9.png)

huayunh commented:
If drawer rail active item can change color to blue 500, then everything looks good style-wise.

![image](https://user-images.githubusercontent.com/8997218/105866908-7b307700-5fc2-11eb-9573-6f9858a73a5d.png)

huayunh commented:
Good work Evan, and thanks for the patience :P

emclaug2 commented:
> If drawer rail active item can change color to blue 500, then everything looks good style-wise.
> 
> ![image](https://user-images.githubusercontent.com/8997218/105866908-7b307700-5fc2-11eb-9573-6f9858a73a5d.png)

This is working in the real dev storybook, so I'm trying to figure out what changed here to cause the theme missing in the rail.

emclaug2 commented:
> This is working in the real dev storybook, so I'm trying to figure out what changed here to cause the theme missing in the rail.

Okay this took some snooping around; here's what's going on:

PX Blue dev ng storybook is not using latest theme @ v5.2.0.  It's using v5.1.2 because the yarn.lock was unchanged, thus preventing dev from using latest theme.  

v5.2.0 introduces a more-specific style rule for applying active nav item font color: 

`
// v5.2.0
.pxb-drawer-nav-item-active .pxb-info-list-item .mat-list-item-content {
    color: map-get($primary, 500);
}
`
I'm not sure what the motivation here was.

This is a bad change; this only applies to _normal_ nav items.  Rail Nav Items are not inside InfoListItems, so this rule doesn't apply to them.

I'm making a new PR in the themes repo that'll fix this using the css below:

`
.pxb-drawer-nav-item-active, /*NavRail item*/
.pxb-drawer-nav-item-active .pxb-info-list-item .mat-list-item-content /*Normal NavItem*/ {
    color: map-get($primary, 500);
}
`

This'll apply the correct active color to rail nav items.

huayunh reviewed (approved):
Change license agreement icon to something that makes more sense
![image](https://user-images.githubusercontent.com/8997218/106025102-71784380-6096-11eb-9183-43aa5445a26c.png)

Looking good otherwise
